students/Net_Michael/lesson5/mailroom3.py

```diff
@@ -0,0 +1,223 @@
+    elif new_donor in donors:
+        new_donor_filename = re.sub(" ", "_", new_donor) + ".txt"
+        donation_directory = os.path.join( donor_filepath, new_donor_filename)
+        new_donation1 = input("Amount of donation by {} ?  ".format(new_donor))
+        new_donation1 = ', ' + str(new_donation1)
+        append_file = open(donation_directory, 'a')
+        append_file.write(new_donation1)
+        append_file.close()
+
+# enter the correct folder name containing all the text files
+# for this computer, the folder name is donor_list
+
+def folder_name_entry():
+    print(" \n Folder name is not correct \n")
+    donor_folder_name = input(" \n Please enter the folder name in your directory \n"
+                                  " containing the list of the donor's names. This \n"
+                                  " folder contains a list of donations in text file \n"
+                                   " named by full name of the donor (firsname_lastname_suffix) \n"
+                                   "\n"
+                                   " Folder name is        ")
+    Donor_folder = donor_list_dir(donor_folder = donor_folder_name)
+    return(Donor_folder)
+
+# switch function to select desired output
+main_sel = {
+        1: enter_donation,
+        2: create_report,
+        3: thank_one_letter,
+        4: thank_all_letter,
+        5: quit_sel,
+        }
+
+prompt = "\n".join(("Welcome",
+          "Please choose from below options: ",
+          "1 - Enter new donation ",
+          "2 - Create a Report ",
+          "3 - Send Thank you to donor ",
+          "4 - Send Thank you to all donors ",
+          "5 - Quit",
+          ">>> "))
+
+# provide the folder name in the machine that contains
+# the donor namelist (example donor_list/Paul_Allen.txt)
+# provide donor_list and it will search accross to find
+# the directorypath */donor_list and will pull the files
+# to execute the desired outputs (report and thank you letters)
+
+if __name__ == "__main__":
+    try:
+        donor_file_trial = donor_list_dir()
+        list_files = os.listdir(donor_file_trial)
+        if(len(list_files) > 0):
+            while True:
+                new_donor_folder = new_donor_folder
+                sel_choice = eval(input(prompt))
+                if sel_choice < 6:
+                    main_sel.get(sel_choice)()
+                else:
+                    print("\n choose less than 4 or press 4 to exit\n")
+    except FileNotFoundError:
+        donor_folder_new = folder_name_entry()
+        list_files = os.listdir(donor_folder_new)
+        folder_new = os.path.basename(donor_folder_new)
+        if(len(list_files) > 0):
+            while True:
+                new_donor_folder = folder_new
+                sel_choice = eval(input(prompt))
+                if sel_choice < 6:
+                    main_sel.get(sel_choice)()
+                else:
+                    print("\n choose less than 4 or press 4 to exit \n")
```

Comment:
@NetMic - unable to run this due to UnboundLocalError.  I'll submit partial credit.  Please submit in a new PR when this is corrected.  Reach out if you have questions.